std/algorithm/iteration.d

```diff
@@ -3660,31 +3660,40 @@ The number of seeds must be correspondingly increased.
 
 // splitter
 /**
-Lazily splits a range using an element as a separator. This can be used with
-any narrow string type or sliceable range type, but is most popular with string
-types.
+Lazily splits a range using an element or element range as a separator.
+This can be used with any narrow string type or sliceable range type,
+but is most popular with string types.
 
 Two adjacent separators are considered to surround an empty element in
 the split range. Use `filter!(a => !a.empty)` on the result to compress
 empty elements.
 
-The predicate is passed to $(REF binaryFun, std,functional), and can either accept
-a string, or any callable that can be executed via `pred(element, s)`.
+The predicate is passed to $(REF binaryFun, std,functional), and accepts
+any callable that can be executed via `pred(element, s)`.
 
-If the empty range is given, the result is an empty range. If a range with
-one separator is given, the result is a range with two empty elements.
+Notes:
+    If splitting a string on whitespace and token compression is desired,
+    consider using `splitter` without specifying a separator.
 
-If splitting a string on whitespace and token compression is desired,
-consider using `splitter` without specifying a separator (see fourth overload
-below).
+    If an element range is used as a separator, it needs to be at least
+    a $(REF_ALTTEXT forward range, isForwardRange, std,range,primitives).
+    Moreover, when a subrange is used a separator, bidirectionality isn't possible.
+
+    If no separator is passed, the $(REF_ALTTEXT, unary, unaryFun, std,functional)
+    predicate `isTerminator` decides whether to accept an element of `r`.
+
+Special_cases:
+    If the empty range is given, the result is an empty range. If a range with
+    one separator is given, the result is a range with two empty elements.
 
 Params:
     pred = The predicate for comparing each element with the separator,
         defaulting to `"a == b"`.
     r = The $(REF_ALTTEXT input range, isInputRange, std,range,primitives) to be
         split. Must support slicing and `.length`.
-    s = The element to be treated as the separator between range segments to be
-        split.
+    s = The element (or element range) to be treated as the separator
+        between range segments to be split.
+    isTerminator = The predicate for deciding where to split the range when no separator is passed
 
 Constraints:
     The predicate `pred` needs to accept an element of `r` and the
```

Comment:
So I tried to group the [endless `splitter` iteration](https://dlang.org/phobos/std_algorithm_iteration.html#.splitter) into one function, but if you think that busts the scope of this PR, I can also move the DDoc unification to another PR.